### DISCUSSION TITLE
custom background changing

### DISCUSSION THREAD

**Discussion by guizinhomartinez** [KadeArchive/Kade-Engine#867] [Q&A]
a lot of mods have the "changing background mid song" thing and i wanted to add into mine
but i have no clue of what i need to do to make it.
can someone help me?

**Comment by Lucky-56** [comment]
short answer: modchart.

long answer: read about modcharts on [the website](https://kadedev.github.io/Kade-Engine/modchart) and have a nice day!

EDIT: I just read the docs and turns out you can't change the stage using a modchart sadly...

**Reply by guizinhomartinez** [reply]
this didn't helped even 1% but thanks i guess

**Reply by Lucky-56** [reply]
yeah sorry, I really thought modcharts could do that

**Reply by epicbobmanbob** [reply]
dont worry. the new 1.7 modchart api has that feature now!

**Comment by ACardboardBox1** [comment]
its simple. add the second bg in the stage code and just make it invisible so it loads with the stage. Then, at the curStep or curBeat (i prefer curStep since you can see it in the charter) where the background changes, make the 2nd bg visible and the first bg invisible. if you want it to change back just do the same thing but swap the roles.
